virtual_workstation_setup.sh

```diff
@@ -23,15 +24,57 @@ popd () {
     command popd "$@" > /dev/null
 }
 
+evg_user() {
+    local evg_username=$(cat $HOME/.evergreen.yml | grep -Po '(?<=user: ).*')
+    if [ -z $evg_username ]; then
+        evg_username=$(cat $HOME/.evergreen.yml | grep -Po '(?<=user:).*')
+    fi
+
+    echo $evg_username
+}
+
+# arg1: file
+# arg2: a unique marker text for the block being appended
+# arg3: block to inject
+# arg4: if non empty, do not append if the file does not exist
+idem_file_append() {
+    if [[ -z $1 ]]; then
+        return 1
+    fi
+    if [[ ! -f $1 && ! -z $4 ]]; then
```

Comment:
Here and elsewhere there a number of shellcheck violations. Some are existing, but might as well clean them up while you're modifying this file. Please ensure this file runs through shellcheck (https://www.shellcheck.net/ or `brew install shellcheck`).